apps/opik-backend/src/test/java/com/comet/opik/infrastructure/http/cors/CorsEnabledE2ETest.java

```diff
@@ -63,13 +67,29 @@ void setUpAll(ClientSupport client) {
     }
 
     @Test
-    @DisplayName("Should contain CORS headers")
-    void testCorsEnabled() {
+    @DisplayName("Should contain CORS headers for GET request")
+    void testCorsHeadersForGet() {
         var response = client.target("%s/is-alive/ping".formatted(baseURI))
-                .request().header("Origin", "localhost")
+                .request().header(HttpHeaders.ORIGIN, "localhost")
                 .get();
 
         assertThat(response.getStatus()).isEqualTo(200);
-        assertThat(response.getHeaders()).containsKey("Access-Control-Allow-Origin");
+        assertThat(response.getHeaders()).containsKey(HttpHeaders.ACCESS_CONTROL_ALLOW_ORIGIN);
+    }
+
+    @Test
+    @DisplayName("Should contain CORS headers for OPTIONS request containing the comet workspace header")
+    void testCorsHeadersForRequestHeader() {
+        try (var response = client.target("%s/v1/private/projects".formatted(baseURI))
+                .request()
+                .header(HttpHeaders.ORIGIN, "localhost")
+                .header(HttpHeaders.ACCESS_CONTROL_REQUEST_HEADERS, CorsFactory.COMET_WORKSPACE_REQUEST_HEADER)
+                .header(HttpHeaders.ACCESS_CONTROL_REQUEST_METHOD, HttpMethod.GET.toString())
+                .options()) {
+            assertThat(response.getStatus()).isEqualTo(200);
+            assertThat(response.getHeaders()).containsKey(HttpHeaders.ACCESS_CONTROL_ALLOW_ORIGIN);
+        } catch (Exception exception) {
+            Assertions.fail(exception);
+        }
```

Comment:
Minor: this code is unnecessary, just allow the exception to bubble and the test will fail as well. Also, you shouldn't use JUnit assertions(use AssertJ instead).

Reply:
Done in https://github.com/comet-ml/opik/pull/520/commits/66dfe0772f9e8a08e96f659e634e26f5263d49c2